src/store/theme-context.jsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import { createContext, useEffect, useState } from 'react'
+
+export const ThemeContext = createContext({
+  theme: 'dark',
+  toggle: () => {},
+})
+
+export default function ThemeProvider({ children }) {
+  const [currentTheme, setCurrentTheme] = useState('light')
+
+  useEffect(() => {
+    checkTheme()
+  }, [currentTheme])
+
+  const checkTheme = () => {
+    const isDarkPreferred = window.matchMedia(
+      '(prefers-color-scheme: dark)'
+    ).matches
+    const isDarkStored = localStorage.theme === 'dark'
+
+    if (isDarkStored || (!localStorage.theme && isDarkPreferred)) {
+      localStorage.setItem('theme', 'dark')
+      document.documentElement.classList.add('dark')
+      setCurrentTheme('dark')
+      return
+    }
+
+    document.documentElement.classList.remove('dark')
+    setCurrentTheme('light')
+  }
+
+  const toggleThemeHandler = () => {
```

Comment:
You can utilize `useCallback` here

Reply:
Please, can you explain why?

Reply:
It prevents unnecessary re-renders. 

https://react.dev/reference/react/useCallback